lib/pos_snapshot_entries.go

```diff
@@ -688,188 +688,169 @@ type SnapshotStakeMapKey struct {
 	}
 
-	// Cache the SnapshotStakeEntries from the db in the UtxoView.
-	for _, snapshotStakeEntry := range dbSnapshotStakeEntries {
-		mapKey := snapshotStakeEntry.ToMapKey()
+	// Cache the snapshot StakeEntries from the db in the UtxoView.
+	for _, stakeEntry := range dbStakeEntries {
+		mapKey := NewSnapshotStakeMapKey(stakeEntry, snapshotAtEpochNumber)
 		if _, exists := bav.SnapshotStakesToReward[*mapKey]; exists {
 			// We should never see duplicate entries from the db that are already in the UtxoView. This is a
-			// sign of a bug and that the utxoViewSnapshotStakeEntries isn't being used correctly.
-			return nil, fmt.Errorf("GetSnapshotStakesToRewardByStakeAmount: db returned a SnapshotStakeEntry" +
+			// sign of a bug and that the utxoViewStakeEntries isn't being used correctly.
+			return nil, fmt.Errorf("GetAllSnapshotStakesToReward: db returned a snapshot StakeEntry" +
 				" that already exists in the UtxoView")
 		}
 
-		bav._setSnapshotStakeToReward(snapshotStakeEntry)
+		bav._setSnapshotStakeToReward(stakeEntry, snapshotAtEpochNumber)
 	}
 
-	// Pull SnapshotStakeEntries from the UtxoView with stake > 0. All entries should have > 0 stake to begin
+	// Pull snapshot StakeEntries from the UtxoView with stake > 0. All entries should have > 0 stake to begin
 	// with, but we filter here again just in case.
-	var mergedSnapshotStakeEntries []*SnapshotStakeEntry
-	for mapKey, snapshotStakeEntry := range bav.SnapshotStakesToReward {
-		if mapKey.SnapshotAtEpochNumber == snapshotAtEpochNumber &&
-			!snapshotStakeEntry.StakeAmountNanos.IsZero() {
-			mergedSnapshotStakeEntries = append(mergedSnapshotStakeEntries, snapshotStakeEntry)
+	var mergedStakeEntries []*StakeEntry
+	for mapKey, stakeEntry := range bav.SnapshotStakesToReward {
+		if mapKey.SnapshotAtEpochNumber == snapshotAtEpochNumber && !stakeEntry.StakeAmountNanos.IsZero() {
+			mergedStakeEntries = append(mergedStakeEntries, stakeEntry)
 		}
 	}
 
-	// Sort the SnapshotStakeEntries DESC by StakeAmountNanos.
-	sort.Slice(mergedSnapshotStakeEntries, func(ii, jj int) bool {
-		stakeAmountCmp := mergedSnapshotStakeEntries[ii].StakeAmountNanos.Cmp(
-			mergedSnapshotStakeEntries[jj].StakeAmountNanos,
-		)
-		if stakeAmountCmp != 0 {
-			return stakeAmountCmp > 0
-		}
-
-		validatorPKIDCmp := bytes.Compare(
-			mergedSnapshotStakeEntries[ii].ValidatorPKID.ToBytes(),
-			mergedSnapshotStakeEntries[jj].ValidatorPKID.ToBytes(),
-		)
-		if validatorPKIDCmp != 0 {
-			return validatorPKIDCmp > 0
-		}
-
-		return bytes.Compare(
-			mergedSnapshotStakeEntries[ii].StakerPKID.ToBytes(),
-			mergedSnapshotStakeEntries[jj].StakerPKID.ToBytes(),
-		) > 0
-	})
-
 	// Return top N.
-	upperBound := limit
-	if uint64(len(mergedSnapshotStakeEntries)) < upperBound {
-		upperBound = uint64(len(mergedSnapshotStakeEntries))
+	upperBound := maxNumSnapshotStakes
+	if uint64(len(mergedStakeEntries)) < upperBound {
+		upperBound = uint64(len(mergedStakeEntries))
 	}
```

Comment:
I'm so giddy about getting min/max functions w/ go 1.21 so we can simplify this at the bottom of every db get function.